### PR TITLE
Remove store km lib and code

### LINKED ISSUE #3123: Remove KM version of eBPF store APIs
As per discussion in the eBPF triage meeting, the following was decided:

- [ ]   UM eBPF store APIs will be exported via ebpfapi.dll. This is covered by #3055 
- [ ] KM eBPF store APIs should be removed. This is being covered by #3109

This issue is to track the above 2 items.

### PR DESCRIPTION
**Pull request by saxena-anurag**
Fixes #3123

## Description

Per discussion in last triage meeting, we need to remove km store libs, and only um store libs will be published from ebpfapi.
This PR removes km store libs.

## Testing

Existing CICD

## Documentation

Updated doc in this PR.

## Installation

No.

### CODE REVIEW

File: docs/eBpfExtensions.md [hunk 1]
```diff
@@ -361,10 +361,9 @@ The eBPF Execution Context loads an eBPF program from an ELF file that has progr
 prefix to these names determines the program type. For example, the section name `"xdp_test"` implies that the corresponding
 program type is `EBPF_PROGRAM_TYPE_XDP_TEST`.
 
-The *Execution Context* discovers the program type associated with a section prefix by reading the data from the ***"eBPF store"***, which is currently kept in the Windows registry.
-When an eBPF extension is installed, it must update the eBPF store with the program types it implements along with the associated section prefixes.
+The *Execution Context* discovers the program type associated with a section prefix by reading the data from the ***"eBPF store"***, which is currently kept in the Windows registry. An extension developer must author a user mode application which will use eBPF store APIs to update the program types it implements along with the associated section prefixes. eBPF store APIs are exported from ebpfapi.dll
```
Comment (dthaler) [inline docs/eBpfExtensions.md:364]:
```suggestion
The *Execution Context* discovers the program type associated with a section prefix by reading the data from the ***"eBPF store"***, which is currently kept in the Windows registry. An extension developer must author a user mode application which will use eBPF store APIs to update the program types it implements along with the associated section prefixes. eBPF store APIs are exported from ebpfapi.dll.
```

File: docs/eBpfExtensions.md [hunk 2]
```diff
@@ -361,10 +361,9 @@ The eBPF Execution Context loads an eBPF program from an ELF file that has progr
...
 
-To operate on the eBPF store, the extension must link the `\lib\ebpf_store_helper_km.lib` kernel-mode library and include the related `\include\ebpf_store_helper.h` header file, both distributed within the [eBPF for Windows NuGet package](https://www.nuget.org/packages/eBPF-for-Windows/). With these, the extension can use the following APIs to register program types, attach types and helper functions:
+To operate on the eBPF store, the user mode application needs to link to eBPFApi.dll and include the related `\include\ebpf_store_helper.h` header file, both distributed within the [eBPF for Windows NuGet package](https://www.nuget.org/packages/eBPF-for-Windows/). With these, the application can use the following APIs to register program types, attach types and helper functions:
```
Comment (dthaler) [inline docs/eBpfExtensions.md:366]:
```suggestion
To operate on the eBPF store, the user mode application needs to link with eBPFApi.dll and include the related `include\ebpf_store_helper.h` header file, both distributed within the [eBPF for Windows NuGet package](https://www.nuget.org/packages/eBPF-for-Windows/). With these, the application can use the following APIs to register program types, attach types, and helper functions:
```

**Review by dthaler — APPROVED** [review]
Looks good, modulo some nits in comments.

**Review by dthaler — APPROVED** [review]

**Comment by gtrevi** [comment]
Should we link this to an issue, to track the new specification end-to-end?

**Comment by saxena-anurag** [comment]
> Should we link this to an issue, to track the new specification end-to-end?

Filed and linked #3123 to this PR.

**Review by gtrevi — APPROVED** [review]

**Review by mtfriesen — APPROVED** [review]

**Pull request merged**